packages/ember-simple-auth/src/services/session.ts

```diff
@@ -334,11 +334,13 @@ export default class SessionService<Data = DefaultDataShape> extends Service {
     @public
   */
   setup() {
-    this._setupIsCalled = true;
-    this._setupHandlers();
-
-    return this.session.restore().catch(() => {
-      // If it raises an error then it means that restore didn't find any restorable state.
-    });
+    if(!this._setupIsCalled) {
+      this._setupIsCalled = true;
+      this._setupHandlers();
+  
+      return this.session.restore().catch(() => {
+        // If it raises an error then it means that restore didn't find any restorable state.
+      });
+    }
```

Comment:
Thanks for the PR :+1: 
It seems like one alternative could be to use an `instance-initializer` instead.

Could you add a test asserting that `_setupHandlers` can be called only once?

We might want to make sure that the setupHandlers runs only once while `restore` continues to be called.
```suggestion
    if (!this._setupIsCalled) {
      this._setupIsCalled = true;
      this._setupHandlers();
    }
      
    return this.session.restore().catch(() => {
      // If it raises an error then it means that restore didn't find any restorable state.
    });
```